recipes-core/initrdscripts/files/init-restore-mode.sh

```diff
@@ -86,10 +100,46 @@ mount_nirecovery_usb()
         if [ ! -d $NIRECOVERY_MOUNTPOINT ]; then
             mkdir -p $NIRECOVERY_MOUNTPOINT
         fi
-        mount -o ro,sync,relatime -L NIRECOVERY $NIRECOVERY_MOUNTPOINT &> /dev/null
+        mount -o ro,sync,relatime UUID=$BOOTMEDIA_UUID $NIRECOVERY_MOUNTPOINT &> /dev/null
     fi
 }
 
+mount_payload()
+{
+   if [ -d $NIRECOVERY_MOUNTPOINT/payload ]; then
+      # We are booting from USB so payload will be on same partition as kernel and initrd
+      mount --bind $NIRECOVERY_MOUNTPOINT/payload /payload;
+   else
+      # We used migration ipk, so payload would be in nirootfs
+      ROOTFS_MOUNTPOINT=$(mktemp -d "/var/volatile/rootfs-XXXXXXX")
+      mount -o rw -L nirootfs $ROOTFS_MOUNTPOINT;
+
+      # Delete everything except payload directory to ensure device
+      # has enough space to extract payload
+      for dir in $ROOTFS_MOUNTPOINT/*; do
+          if [ ${dir##*/} != "payload" ]; then
+              rm -rf $dir;
```

Comment:
If `mktemp` fails for whatever reason (like `/var/volatile` being mounted read-only), `ROOTFS_MOUNTPOINT` will have a null value, and line 121 will `rm -rf` everything from root which isn't `/payload`.

It's unlikely that `mktemp` will fail like that, but can you add a check to keep it from being so destructive?

Reply:
Done. Thanks for reviewing